kalebswartz7/excercise_01/parse.py

```diff
@@ -0,0 +1,136 @@
+# Parse the data from the dat file -> Points vs Horizontal Position = how many rows per column in the 2D array 
+#                                  -> Points vs Vertical Position = how many columns in the 2D array 
+
+#       Create a dictionary and populate it with other dictionaries based on the data in the dat file 
+#       Use this data to add to the graph
+
+import matplotlib as mpl
+import numpy as np
+import matplotlib.pyplot as plt
+
+#Just default values, these are changed based on the dat file
+horizontal = 1 
+vertical = 1
+
+#initialize the dictionary for storing parameters given in the dat file 
+initialData = {}
+initialData['photon'] = {}
+initialData['horizontal'] = {}
+initialData['vertical'] = {}
+matrix = [[]] 
+
+""" 
+Get data from the dat file with the parameters for the scan. Put this data into an organized Dictionary
+
+"""
+def getCommentedData():
+    with open('res_int_pr_se.dat') as file:
+        lineNum = 1
+        for line in file:
```

Comment:
You can use `contents = file.readlines()` for small files (a few MB), or `file.readline()` with a loop of known length (10-20 header lines).

Reply:
Please use this as a reference:
```py
In [24]: content = []
    ...: with open('res_int_pr_se.dat', 'r') as f:
    ...:     for i in range(20):
    ...:         content.append(f.readline())
    ...:

In [25]:

In [25]: content = [x.strip() for x in content]

In [26]: content
Out[26]:
['# [ph/s/.1%bw/mm^2] (C-aligned, inner loop is vs Photon Energy, outer loop vs Vertical Position)',
 '#9000.0 #Initial Photon Energy [eV]',
 '#9000.0 #Final Photon Energy [eV]',
 '#1 #Number of points vs Photon Energy',
 '#-0.00014237188284963375 #Initial Horizontal Position [m]',
 '#0.00014256428526195097 #Final Horizontal Position [m]',
 '#528 #Number of points vs Horizontal Position',
 '#-6.997475022319695e-05 #Initial Vertical Position [m]',
 '#6.99747502231969e-05 #Final Vertical Position [m]',
 '#294 #Number of points vs Vertical Position',
 '88871.125',
 '90252.6640625',
 '94020.2265625',
 '96684.3203125',
 '96943.7421875',
 '98506.265625',
 '102780.25',
 '106489.9453125',
 '108295.4140625',
 '109468.7734375']

In [27]: content = [x.strip() for x in content if x.startswith('#')]

In [28]: content
Out[28]:
['# [ph/s/.1%bw/mm^2] (C-aligned, inner loop is vs Photon Energy, outer loop vs Vertical Position)',
 '#9000.0 #Initial Photon Energy [eV]',
 '#9000.0 #Final Photon Energy [eV]',
 '#1 #Number of points vs Photon Energy',
 '#-0.00014237188284963375 #Initial Horizontal Position [m]',
 '#0.00014256428526195097 #Final Horizontal Position [m]',
 '#528 #Number of points vs Horizontal Position',
 '#-6.997475022319695e-05 #Initial Vertical Position [m]',
 '#6.99747502231969e-05 #Final Vertical Position [m]',
 '#294 #Number of points vs Vertical Position']

```

Reply:
Has been fixed.